src/sortFields.js

```diff
@@ -0,0 +1,393 @@
+// Taken from project marc-record-js, file marcSortFields.js as this contains more and more Melinda-specific rules.
+
+import clone from 'clone';
+import createDebugLogger from 'debug';
+import {fieldHasSubfield, fieldToString} from './utils';
+import {isValidSubfield8} from './subfield8Utils';
+import {isValidSubfield6, subfield6GetOccurrenceNumber} from './subfield6Utils';
+
+
+const BIG_BAD_NUMBER = 999.99;
+
+const debug = createDebugLogger('@natlibfi/marc-record-validators-melinda:sortFields');
+//const debugData = debug.extend('data');
+const debugDev = debug.extend('dev');
+
+export default function () {
+
+  return {
+    description: 'Sort fields',
+    validate, fix
+  };
+
+  function fix(record) {
+    const res = {message: [], fix: [], valid: true};
+
+    record.fields.sort(fieldOrderComparator); // eslint-disable-line functional/immutable-data
+
+    return res;
+  }
+
+  function validate(record) {
+    const res = {message: []};
+
+    const fields = record.fields.map(f => clone(f));
+    fields.sort(fieldOrderComparator); // eslint-disable-line functional/immutable-data
+
+
+    const relocatedFields = fields.filter((f, i) => fieldToString(f) !== fieldToString(record.fields[i]));
+
+    if (relocatedFields.length > 0) { // eslint-disable-line functional/no-conditional-statements
+      res.message.push(`${relocatedFields.length} field(s) in new places`); // eslint-disable-line functional/immutable-data
+    }
+
+    res.valid = !(res.message.length >= 1); // eslint-disable-line functional/immutable-data
+    return res;
+  }
+}
+
+
+export const relatorTermScore = { // Here bigger is better
+  // NB! This is exportable as field internal $e sorting in marc-record-validators-js uses this.
+  // NB! The more abstract, the earlier it appears.
+  // Note that terms with same abstraction level might also have order preferences
+  // We should 1) check the order of these, and 2) add translations (support Swedish at the very least)
+  // work/teos > expression/ekspressio > manifestation/manifestaatio
+  'säveltäjä': 100, 'composer': 100,
+  'kirjoittaja': 99, 'author': 100,
+  'sarjakuvantekijä': 99,
+  'taiteilija': 98,
+  'sanoittaja': 90,
+  'käsikirjoittaja': 90,
+  // expression:
+  'toimittaja': 80, 'editor': 80,
+  'sovittaja': 80, 'arranger': 80,
+  'kuvittaja': 75,
+  'editointi': 71, // for music, editor/toimittaja is another thing
+  'kääntäjä': 70,
+  'lukija': 61,
+  // Manifestation level
+  'esittäjä': 60,
+  'johtaja': 50, // orkesterinjohtaja
+  'kustantaja': 41,
+  'julkaisija': 40
+
+};
+
+export function scoreRelatorTerm(value) { // sortRelatorTerms.js validator should call this on future version
+  const normValue = value.replace(/[.,]+$/u, '');
+  if (normValue in relatorTermScore) {
+    return relatorTermScore[normValue];
+  }
+  return 0;
+}
+
+export function fieldOrderComparator(fieldA, fieldB) {
```

Comment:
Because we don't want Finland / Melinda specific sorting (for example sorting Finnish vocabularies first for subject fields) to marc-record-js.

Reply:
Echoing Minttu... This intermediate validator is essentially a copy-paste of the original. Only after this goes to production, the original marc-record-js' sorting is simplified to it's "final version" (= melinda-specific conventions etc are removed) and it will have only:

`const sorterFunctions = [sortByTag, sortAlphabetically];`

Eventually there will either be fieldOrderComparator or something in both projects OR sorterFunction array will be passed as a function parameter to marc-record-js' fieldOrderComparator function (probably latter). The validator will use something like

`const sorterFunctions = [sortByTag, sortByIndexTerms, sortAlphabetically, preferFenniKeep];`

where sortByTag and sortAlphabetically are exported from marc-record-js.